### PR TITLE
Add cert, y2038 and threadsafety addons

### LINKED ISSUE #9: false positives in cppcheck ... fixed in 1.83?
I'm noticing that cppcheck seems lots better in 1.83, eliminating a really annoying false positive.  Please condider updating.

### LINKED ISSUE #29: Add cert, y2038 and threadsafety addons

### PR DESCRIPTION
**Pull request by ljmf00**
Fixes #29 .

### CODE REVIEW

**Comment by lolgab** [comment]
Use our convention for commit messages.

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,5 @@
 .DS_Store
 .metals/
 .bloop/
+
+*.dump
```
Comment (lolgab) [inline .gitignore]: Newline here

File: src/main/resources/docs/description/description.json [hunk 2]
```diff
@@ -1453,4 +1453,68 @@
...
+  "timeToFix" : 5
+}, {
+  "patternId" : "y2038-type-bits-undef",
+  "title" : "_USE_TIME_BITS64 is defined but _TIME_BITS was not",
+  "timeToFix" : 5
+}, {
+  "patternId" : "y2038-unsafe-call",
+  "title" : "Using Y2038-unsafe call",
+  "timeToFix" : 5
 } ]
```
Comment (lolgab) [inline src/main/resources/docs/description/description.json]: Tweak the DocGenerator to put a newline here.

File: src/main/resources/addons/patterns.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
+{
+  "patterns" : [
+    {
+      "patternId" : "threadsafety-threadsafety",
+      "level" : "Warning",
+      "category" : "ErrorProne"
+    },
```
Comment (lolgab) [inline src/main/resources/addons/patterns.json]: Why do you need this file?
It is not read by Codacy, and you have already the pattern inside the main `pattern.json` file.

Comment (ljmf00) [reply src/main/resources/addons/patterns.json]: This is the manual documentation of addons. With addons, we can't generate documentation automatically, unfortunately. So this file will then be merged with the automatic generated files.

File: .gitignore (hunk 1 above)
Comment (ljmf00) [reply .gitignore]: Done

File: src/main/resources/docs/description/description.json (hunk 2 above)
Comment (ljmf00) [reply src/main/resources/docs/description/description.json]: Done

**Comment by ljmf00** [comment]
> Use our convention for commit messages.

I tried to reword those according to the handbook. Check now. Let me know if there's anything more

File: src/main/resources/addons/patterns.json (hunk 3 above)
Comment (lolgab) [reply src/main/resources/addons/patterns.json]: Should these files be added to the tool image? I don't know if there is a better way to add them

Comment (ljmf00) [reply src/main/resources/addons/patterns.json]: No the files are not added to the image, only the final files, inside `docs/`.

Comment (gjsduarte) [reply src/main/resources/addons/patterns.json]: I think it makes sense to have a base file for manual patterns, and then to join with patterns.json when generating the documentation. 👍

**Pull request merged**